### PR TITLE
s/coin/token/g

### PR DESCRIPTION
**Pull request by cap10morgan**
Update to replace all instances of "coin" with "token" & an implementation switch to the leaves data structure (from map to array).

### CODE REVIEW

File: docs/whitepaper.md [hunk 1]
```diff
@@ -83,72 +83,72 @@ Only the owner(s) of a Chain Tree may create new transactions on a Chain Tree. T
...
-RECEIVE_COIN
-message ReceiveCoinTransaction {
-   string send_coin_transaction_id = 1;
+RECEIVE_TOKEN
+message ReceiveTokenTransaction {
+   string send_token_transaction_id = 1;
    bytes tip = 2;
    signature signature = 3;
-   map<string>ChainTreeNode leaves = 4;
+   array<ChainTreeNode> leaves = 4;
```
Comment (aknuds1) [inline docs/whitepaper.md:141]: I presume the change from `map` to `array` is intended?

**Review by aknuds1 — APPROVED** [review]
LGTM!

File: docs/whitepaper.md [hunk 2]
```diff
@@ -83,72 +83,72 @@ Only the owner(s) of a Chain Tree may create new transactions on a Chain Tree. T
...
+Any actor may mint their own tokens once they have been established within that chaintree via an `ESTABLISH_TOKEN` transaction. They do so by creating a MINT transaction on their Chain Tree. In order for a `MINT_TOKEN` transaction to be valid, the name of the currency must be prefaced by their Chain Tree’s DID. An example cryptocurrency might be named: `did:tupelo:0xF964A90A0be7aC039E415aed6b2DD97651316700:cat_token` where `did:tupelo:0xF964A90A0be7aC039E415aed6b2DD97651316700` is the DID and \'cat_token\' is the name of the currency. The only potential limitation a Chain Tree owner has on minting more of their own currency is the Monetary Policy set in place when the token was established.
```
Comment (zonotope) [inline docs/whitepaper.md]: Should we also change "currency" to token here? 
> An example *token* might be named ...

and

> ... and \'cat_token\' is the name of the *token*

**Review by zonotope — APPROVED** [review]
It looks good to me besides a minor correction that I don't feel strongly about and would be just fine if it was ignored.

**Review by tobowers — APPROVED** [review]
Lgtm. I like Ben’s comment too

**Pull request merged**